### PR TITLE
Avoid two more sleep() calls in the pdfork tests and fix a race

### PR DESCRIPTION
**Pull request by arichardson**
As with the previous commit use a pipe for startup synchronization.
The PollMultiple test can also sometimes fail on slow systems.
Synchronizing the child+parent states via a pipe avoids this problem.

### CODE REVIEW

**Comment by arichardson** [comment]
rebased on top of #49

File: procdesc.cc [hunk 1]
```diff
@@ -384,12 +406,18 @@ TEST_F(PipePdfork, PollMultiple) {
...
   EXPECT_EQ(0, poll(&fdp, 1, 0));
 
+  // Allow C to exit:
+  if (verbose) fprintf(stderr, "[%d] telling child to exit\n", getpid_());
+  SEND_INT_MESSAGE(pipefds[0], MSG_PARENT_REQUEST_CHILD_EXIT);
+  close(pipefds[0]);
+
   // Now, wait (indefinitely) for activity on the process descriptor.
   // We expect:
   //  - pid C will finish its sleep, write to the pipe and exit
```
Comment (daviddrysdale) [inline procdesc.cc]: Does this comment need an update?

File: procdesc.cc [hunk 2]
```diff
@@ -109,27 +109,39 @@ void CheckChildFinished(pid_t pid, bool signaled=false) {
```
Comment (daviddrysdale) [inline procdesc.cc:109]: BTW, there's a TODO near the top of this file (~l.30) that you can probably remove when you're done &ndash; thanks!

File: procdesc.cc [hunk 3]
```diff
@@ -384,12 +406,18 @@ TEST_F(PipePdfork, PollMultiple) {
   fdp.revents = 0;
   EXPECT_EQ(0, poll(&fdp, 1, 0));
 
+  // Allow C to exit:
```
Comment (daviddrysdale) [inline procdesc.cc]: Maybe "Both A and D ask C to exit, allowing it to do so."

**Review by daviddrysdale — COMMENTED** [review]
Just a couple of tiny clarifications to make the test easier to follow, thanks.

File: procdesc.cc (hunk 2 above)
Comment (arichardson) [reply procdesc.cc:109]: There are still a few tests that use sleep, I'll try to fix those too and then remove the comment.

Comment (arichardson) [reply procdesc.cc:109]: I'll open a separate PR for those remainig tests though.

**Pull request merged**